### PR TITLE
fix: mitigate flakiness of TestBasicWakuV2

### LINKED ISSUE #4105: Flaky test - TestBasicWakuV2
```
0x3af229605178b48da99111097234feeb7b4e08c96ec50cfbe339ad862a6839bd", "receivedTime": 1696507482315651441, "payloadSizeBytes": 796}
2023-10-05T12:04:42.315Z	DEBUG	wakuv2/waku.go:1337	cached w envelope	{"envelopeHash": "0x3af229605178b48da99111097234feeb7b4e08c96ec50cfbe339ad862a6839bd"}
2023-10-05T12:04:42.315Z	DEBUG	wakuv2/waku.go:1369	filters did not match	{"hash": "0x3af229605178b48da99111097234feeb7b4e08c96ec50cfbe339ad862a6839bd", "contentTopic": "/waku/1/0xfa61c2f2/rfc26"}
    waku_test.go:162: 
        	Error Trace:	/home/jenkins/workspace/status-go_prs_tests_PR-4070/wakuv2/waku_test.go:162
        	Error:      	Should not be zero, but was 0
        	Test:       	TestBasicWakuV2
```

### PR DESCRIPTION
**Pull request by osmaczko**
fixes: #4105

### CODE REVIEW

File: wakuv2/waku_test.go [hunk 1]
```diff
@@ -158,9 +158,17 @@ func TestBasicWakuV2(t *testing.T) {
 	require.Len(t, messages, 1)
 
 	timestampInSeconds := msgTimestamp / int64(time.Second)
-	storeResult, err := w.query(context.Background(), storeNode.PeerID, relay.DefaultWakuTopic, []common.TopicType{contentTopic}, uint64(timestampInSeconds-20), uint64(timestampInSeconds+20), []store.HistoryRequestOption{})
-	require.NoError(t, err)
-	require.NotZero(t, len(storeResult.Messages))
+
+	querySucceeded := false
+	for i := 0; i < 3; i++ { // Naive way to mitigate the flakiness of `w.query`
```
Comment (jrainville) [inline wakuv2/waku_test.go]: Could you make this a TODO to remove the loop when the test is fixed for real

**Review by jrainville — APPROVED** [review]
Looks good, I'd just add a TODO

**Comment by osmaczko** [comment]
~~Converting to draft as it seems to not improve the stability~~ :thinking:

**Comment by osmaczko** [comment]
@richard-ramos could you please take a look? Do you think it makes sense?

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**